pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -355,13 +355,15 @@ func (p *ldapProvider) searchUser(name string, config *v3.LdapConfig, lConn *lda
 	for _, attr := range srchAttributes {
 		srchAttrs += fmt.Sprintf("(%v=%v*)", attr, name)
 	}
-	query += srchAttrs + "))"
+	// The user search filter will be added as another and clause
+	// and is expected to follow ldap syntax and enclosed in parenthesis
+	query += srchAttrs + ")" + config.UserSearchFilter + ")"
```

Comment:
di we validate the searchFilter input?

Reply:
We don't validate it for AD either. If we want to validate, then during auth setup we can run ldapsearch with this filter, if it errors then we don't setup auth?

Reply:
so seems like we let configure their own search filter, we should document it that admin needs to enter a valid filter.